### DISCUSSION TITLE
useEffect is not working in production mode

### DISCUSSION THREAD

**Discussion by straiffix** [vercel/next.js#48754] [Help]
### Summary

Hi Community! 

I have a problem with fetching posts from firebase. It seems working in development mode, but in production nothing shows up. It looks like problem is in useEffect. Any ideas why this isn't working in production? Thank you in advance.

code: https://github.com/straiffix/fractumscrypt

```

import Head from 'next/head';
import { db } from "../components/firebase"
import { onValue, ref } from "firebase/database";
import { collection, getDocs } from "firebase/firestore";
import { useEffect } from 'react';
import ShortPost from '../components/ShortPost'
import { useState } from 'react';
import Layout from '../layouts/Layout';
import { useRouter } from "next/router";


export default function Home() {

  const [posts, setPosts] = useState([])

  useEffect(()=>{
    async function fetchPosts() {
       
      const fetchedPosts = await getDocs(collection(db, "posts"));
      const newData = fetchedPosts.docs.map((doc) => 
                {return <ShortPost  key={doc.id} post_id = {doc.id} post ={doc.data()} /> }
                );   
      setPosts(newData); 
      }
    fetchPosts()
    }, [posts])

  return (
    <Layout home>
        {posts}
    </Layout>
  );
  
}
```

### Example

https://github.com/straiffix/fractumscrypt/blob/main/pages/index.js

**Comment by Hamzi-SE** [comment]
Based on your code, it seems that the useEffect hook is causing an infinite loop. This is because you have set the dependency of the useEffect hook as posts. Whenever the posts state changes, the useEffect hook is triggered again, causing another fetch of the posts data, which then updates the posts state again, triggering the useEffect hook again, and so on. This loop continues indefinitely, causing the app to hang and preventing it from rendering anything.

To fix this, you should remove the posts dependency from the useEffect hook and move the fetchPosts function outside of the hook. This way, the fetchPosts function will only be called once, when the component mounts, and not on every state change.

````
 const [posts, setPosts] = useState([]);

  async function fetchPosts() {
    const fetchedPosts = await getDocs(collection(db, "posts"));
    const newData = fetchedPosts.docs.map((doc) => {
      return <ShortPost key={doc.id} post_id={doc.id} post={doc.data()} />
    });   
    setPosts(newData); 
  }

  useEffect(() => {
    fetchPosts();
  }, []);
````

**Reply by straiffix** [reply]
I tried, but it still not working, no posts show up.

**Reply by Hamzi-SE** [reply]
Check the console and network tab to see if you are getting any error

**Reply by straiffix** [reply]
It looks like there is no any requests. There is no any build errors also. 

![Screenshot_20230423_184301](https://user-images.githubusercontent.com/24612656/233853342-f7ef5138-49cd-42bb-abde-3964226a167f.png)
![Screenshot_20230423_184332](https://user-images.githubusercontent.com/24612656/233853349-a57e3041-9e93-4a25-83ae-7564e5391bbe.png)

**Comment by straiffix** [comment]
Solved, I am not sure why but the problem was in next.config.js with following content:
```
const isProd = process.env.NODE_ENV === 'production'

module.exports = {
  assetPrefix: isProd ? '/fractumscrypt/' : '',
  images: {
    unoptimized: true,
  },
}
```

After removing the file everything started to work.
